examples/booleansearchparser.py

```diff
@@ -0,0 +1,488 @@
+        )
+        
+        operatorOr << (Group(
+            operatorAnd + Suppress(Keyword("or", caseless=True)) + operatorOr
+        ).setResultsName("or") | operatorAnd)
+
+        return operatorOr.parseString
+    def evaluateAnd(self, argument):
+        return self.evaluate(argument[0]) and (self.evaluate(argument[1]))
+        
+    def evaluateOr(self, argument):
+        return self.evaluate(argument[0]) or self.evaluate(argument[1])
+        
+
+    def evaluateNot(self, argument):
+        return self.GetNot(self.evaluate(argument[0]))
+
+    def evaluateParenthesis(self, argument):
+        return self.evaluate(argument[0])
+    
+    def evaluateQuotes(self, argument):
+        """Evaluate quoted strings
+
+        First is does an 'and' on the indidual search terms, then it asks the
+        function GetQuoted to only return the subset of ID's that contain the
+        literal string.
+        """
+        #r = Set()
+        r = False
+        search_terms = []
+        for item in argument:
+            search_terms.append(item[0])
+            r = r and self.evaluate(item)
+        return self.GetQuotes(' '.join(search_terms), r)
+    
+    def evaluateWord(self, argument):
+        wildcard_count = argument[0].count(u"*")
+        if wildcard_count > 0:
+            if wildcard_count == 1 and argument[0].startswith(u"*"):
+                return self.GetWordWildcard(argument[0][1:], method = "endswith")
+            if wildcard_count == 1 and argument[0].endswith(u"*"):
+                return self.GetWordWildcard(argument[0][:-1], method = "startswith")
+            else:
+                _regex  = argument[0].replace(u"*",u".+")
+                matched = False
+                for w in self.words:
+                    matched = bool(re.search(_regex,w))
+                    if matched:
+                        break
+                return matched
+        
+        return self.GetWord(argument[0])
+
+    def evaluateWordWildcardPrefix(self, argument):
+        return self.GetWordWildcard(argument[0], method = "endswith")
+    
+    def evaluateWordWildcardSufix(self, argument):
+        return self.GetWordWildcard(argument[0], method = "startswith")
+    
+    def evaluate(self, argument):
+        return self._methods[argument.getName()](argument)
+
+    def Parse(self, query):
+        return self.evaluate(self._parser(query)[0])
+
+    def GetWord(self, word):
+        return word in self.words
+
+    def GetWordWildcard(self, word, method = "startswith"):
+        matched = False
+        for w in self.words:
+            matched = getattr(w,method)(word)
+            if matched:
+                break
+        return matched
+    
+    """
+    def GetKeyword(self, name, value):
+        return Set()
+
+    def GetBetween(self, min, max):
+        print (min,max)
+        return Set()
+    """
+    
+    def GetQuotes(self, search_string, tmp_result):
+        return self.text.count(search_string) > 0
```

Comment:
Better written as:

    return search_string in self.text

count() will scan the entire string, when it is only necessary to scan to the first occurrence.